.circleci/config.yml

```diff
@@ -670,7 +699,18 @@ workflows:
           requires:
             - prepare_patch_release_approval
 
+  nightly:
+    when: << pipeline.parameters.nightly >>
+    jobs:
+      - build_release:
+          matrix:
+            parameters:
+              platform:
+                [macos_build, windows_build, amd_linux_build, arm_linux_build]
+
```

Comment:
Note that this doesn't currently actually publish anything to GitHub Releases like normal releases.  Assuming these jobs don't start failing immediately, my next steps are to introduce a step that saves these to CircleCI artifacts:

```
      - store_artifacts:
          path: ./artifacts
```

Roughly documented [here](https://circleci.com/docs/artifacts/#uploading-artifacts).